### ISSUE TITLE
Error when you try to withdraw 0 from wallet with zero balance 

### ISSUE THREAD

**Issue by gfedorenko** [openware/peatio#130]
Need to handle it nicely

![screenshot from 2017-12-19 17-55-32](https://user-images.githubusercontent.com/25289193/34165855-19d0e706-e4e6-11e7-90ce-6733163e2986.png)
![screenshot from 2017-12-19 17-56-01](https://user-images.githubusercontent.com/25289193/34165861-1d75375e-e4e6-11e7-88be-f91143ffb6cf.png)

```
NoMethodError at /withdraws/satoshis ==================================== > undefined method `[]=' for nil:NilClass app/controllers/concerns/withdraws/withdrawable.rb, line 33 ----------------------------------------------------------- ``` ruby 28 @fund_sources = current_user.fund_sources.with_currency(channel.currency) 29 @assets = model_kls.without_aasm_state(:submitting).where(member: current_user).order(:id).reverse_order.limit(10) 30 end 31 32 def withdraw_params > 33 params[:withdraw][:currency] = channel.currency 34 params[:withdraw][:member_id] = current_user.id 35 params.require(:withdraw).permit(:fund_source_id, :member_id, :currency, :sum) 36 end 37 38 end ``` App backtrace ------------- - app/controllers/concerns/withdraws/withdrawable.rb:33:in `withdraw_params' - app/controllers/concerns/withdraws/withdrawable.rb:10:in `create' - lib/middleware/security.rb:11:in `call' - lib/middleware/i18n_js.rb:9:in `call' Full backtrace -------------- - app/controllers/concerns/withdraws/withdrawable.rb:33:in `withdraw_params' - app/controllers/concerns/withdraws/withdrawable.rb:10:in `create' - actionpack (4.2.10) lib/action_controller/metal/implicit_render.rb:4:in `send_action' - actionpack (4.2.10) lib/abstract_controller/base.rb:198:in `process_action' - actionpack (4.2.10) lib/action_controller/metal/rendering.rb:10:in `process_action' - actionpack (4.2.10) lib/abstract_controller/callbacks.rb:20:in `block in process_action' - activesupport (4.2.10) lib/active_support/callbacks.rb:117:in `call' - activesupport (4.2.10) lib/active_support/callbacks.rb:555:in `block (2 levels) in compile' - activesupport (4.2.10) lib/active_support/callbacks.rb:505:in `call' - activesupport (4.2.10) lib/active_support/callbacks.rb:92:in `__run_callbacks__' - activesupport (4.2.10) lib/active_support/callbacks.rb:778:in `_run_process_action_callbacks' - activesupport (4.2.10) lib/active_support/callbacks.rb:81:in `run_callbacks' - actionpack (4.2.10) lib/abstract_controller/callbacks.rb:19:in `process_action' - actionpack (4.2.10) lib/action_controller/metal/rescue.rb:29:in `process_action' - actionpack (4.2.10) lib/action_controller/metal/instrumentation.rb:32:in `block in process_action' - activesupport (4.2.10) lib/active_support/notifications.rb:164:in `block in instrument' - activesupport (4.2.10) lib/active_support/notifications/instrumenter.rb:20:in `instrument' - activesupport (4.2.10) lib/active_support/notifications.rb:164:in `instrument' - actionpack (4.2.10) lib/action_controller/metal/instrumentation.rb:30:in `process_action' - actionpack (4.2.10) lib/action_controller/metal/params_wrapper.rb:250:in `process_action' - activerecord (4.2.10) lib/active_record/railties/controller_runtime.rb:18:in `process_action' - actionpack (4.2.10) lib/abstract_controller/base.rb:137:in `process' - actionview (4.2.10) lib/action_view/rendering.rb:30:in `process' - actionpack (4.2.10) lib/action_controller/metal.rb:196:in `dispatch' - actionpack (4.2.10) lib/action_controller/metal/rack_delegation.rb:13:in `dispatch' - actionpack (4.2.10) lib/action_controller/metal.rb:237:in `block in action' - actionpack (4.2.10) lib/action_dispatch/routing/route_set.rb:74:in `dispatch' - actionpack (4.2.10) lib/action_dispatch/routing/route_set.rb:43:in `serve' - actionpack (4.2.10) lib/action_dispatch/journey/router.rb:43:in `block in serve' - actionpack (4.2.10) lib/action_dispatch/journey/router.rb:30:in `serve' - actionpack (4.2.10) lib/action_dispatch/routing/route_set.rb:817:in `call' - omniauth (1.2.2) lib/omniauth/strategy.rb:186:in `call!' - omniauth (1.2.2) lib/omniauth/strategy.rb:164:in `call' - omniauth (1.2.2) lib/omniauth/strategy.rb:186:in `call!' - omniauth (1.2.2) lib/omniauth/strategy.rb:164:in `call' - omniauth (1.2.2) lib/omniauth/strategy.rb:404:in `call_app!' - omniauth-identity (1.1.1) lib/omniauth/strategies/identity.rb:43:in `other_phase' - omniauth (1.2.2) lib/omniauth/strategy.rb:185:in `call!' - omniauth (1.2.2) lib/omniauth/strategy.rb:164:in `call' - omniauth (1.2.2) lib/omniauth/builder.rb:59:in `call' - meta_request (0.4.3) lib/meta_request/middlewares/app_request_handler.rb:13:in `call' - meta_request (0.4.3) lib/meta_request/middlewares/meta_request_handler.rb:13:in `call' - http_accept_language (2.1.1) lib/http_accept_language/middleware.rb:14:in `call' - simple_captcha2 (0.4.3) lib/simple_captcha/middleware.rb:47:in `call' - rack (1.6.8) lib/rack/etag.rb:24:in `call' - rack (1.6.8) lib/rack/conditionalget.rb:38:in `call' - rack (1.6.8) lib/rack/head.rb:13:in `call' - actionpack (4.2.10) lib/action_dispatch/middleware/params_parser.rb:27:in `call' - actionpack (4.2.10) lib/action_dispatch/middleware/flash.rb:260:in `call' - rack (1.6.8) lib/rack/session/abstract/id.rb:225:in `context' - rack (1.6.8) lib/rack/session/abstract/id.rb:220:in `call' - actionpack (4.2.10) lib/action_dispatch/middleware/cookies.rb:560:in `call' - activerecord (4.2.10) lib/active_record/query_cache.rb:36:in `call' - activerecord (4.2.10) lib/active_record/connection_adapters/abstract/connection_pool.rb:653:in `call' - activerecord (4.2.10) lib/active_record/migration.rb:377:in `call' - actionpack (4.2.10) lib/action_dispatch/middleware/callbacks.rb:29:in `block in call' - activesupport (4.2.10) lib/active_support/callbacks.rb:88:in `__run_callbacks__' - activesupport (4.2.10) lib/active_support/callbacks.rb:778:in `_run_call_callbacks' - activesupport (4.2.10) lib/active_support/callbacks.rb:81:in `run_callbacks' - actionpack (4.2.10) lib/action_dispatch/middleware/callbacks.rb:27:in `call' - actionpack (4.2.10) lib/action_dispatch/middleware/reloader.rb:73:in `call' - actionpack (4.2.10) lib/action_dispatch/middleware/remote_ip.rb:78:in `call' - better_errors (2.4.0) lib/better_errors/middleware.rb:84:in `protected_app_call' - better_errors (2.4.0) lib/better_errors/middleware.rb:79:in `better_errors_call' - better_errors (2.4.0) lib/better_errors/middleware.rb:57:in `call' - actionpack (4.2.10) lib/action_dispatch/middleware/debug_exceptions.rb:17:in `call' - rack-contrib (1.8.0) lib/rack/contrib/response_headers.rb:17:in `call' - meta_request (0.4.3) lib/meta_request/middlewares/headers.rb:16:in `call' - actionpack (4.2.10) lib/action_dispatch/middleware/show_exceptions.rb:30:in `call' - railties (4.2.10) lib/rails/rack/logger.rb:38:in `call_app' - railties (4.2.10) lib/rails/rack/logger.rb:20:in `block in call' - activesupport (4.2.10) lib/active_support/tagged_logging.rb:68:in `block in tagged' - activesupport (4.2.10) lib/active_support/tagged_logging.rb:26:in `tagged' - activesupport (4.2.10) lib/active_support/tagged_logging.rb:68:in `tagged' - railties (4.2.10) lib/rails/rack/logger.rb:20:in `call' - quiet_assets (1.1.0) lib/quiet_assets.rb:27:in `call_with_quiet_assets' - request_store (1.3.2) lib/request_store/middleware.rb:9:in `call' - actionpack (4.2.10) lib/action_dispatch/middleware/request_id.rb:21:in `call' - rack (1.6.8) lib/rack/methodoverride.rb:22:in `call' - rack (1.6.8) lib/rack/runtime.rb:18:in `call' - lib/middleware/security.rb:11:in `call' - rack (1.6.8) lib/rack/lock.rb:17:in `call' - actionpack (4.2.10) lib/action_dispatch/middleware/static.rb:120:in `call' - lib/middleware/i18n_js.rb:9:in `call' - rack (1.6.8) lib/rack/sendfile.rb:113:in `call' - railties (4.2.10) lib/rails/engine.rb:518:in `call' - railties (4.2.10) lib/rails/application.rb:165:in `call' - rack (1.6.8) lib/rack/content_length.rb:15:in `call' - puma (3.11.0) lib/puma/configuration.rb:225:in `call' - puma (3.11.0) lib/puma/server.rb:624:in `handle_request' - puma (3.11.0) lib/puma/server.rb:438:in `process_client' - puma (3.11.0) lib/puma/server.rb:302:in `block in run' - puma (3.11.0) lib/puma/thread_pool.rb:120:in `block in spawn_thread'
```

**Comment by jinnysunny** [comment]
@gfedorenko this happens to you because the bitcoin daemon is not connected. You should check that it's running and that the user/pass combo is correct.

Correction:
My bad it's not related.

**Comment by yivo** [comment]
Closed in #288.
